lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -158,13 +162,14 @@ impl<TInvertedIndex: InvertedIndex> VectorIndex for SparseVectorIndex<TInvertedI
         let borrowed_id_tracker = self.id_tracker.borrow();
         let deleted_bitslice = borrowed_vector_storage.deleted_vector_bitslice();
         let mut ram_index = InvertedIndexRam::empty();
-        let available_point_count = borrowed_id_tracker.available_point_count();
+        let mut index_point_count: usize = 0;
         for id in borrowed_id_tracker.iter_ids_excluding(deleted_bitslice) {
             check_process_stopped(stopped)?;
             let vector: &SparseVector = borrowed_vector_storage.get_vector(id).try_into()?;
             ram_index.upsert(id, vector.to_owned());
+            index_point_count += 1;
         }
-        self.max_point_id = available_point_count as PointOffsetType - 1;
+        self.max_point_id = index_point_count.saturating_sub(1) as PointOffsetType;
```

Comment:
protect from overflow if zero point

Reply:
Or we can keep count of points instead of max index. We use this field to return the count of indexed points. And increment this counter in update if `Self::update_vector` added new vector